nx/guides/modoc_agg_nx_01.livemd

```diff
@@ -0,0 +1,826 @@
+\end{bmatrix}
+\\ 
+\begin{bmatrix} 
+  & 2, 1 & \\
+  & 0, \boxed{1} & \\
+  & 1, 0
+\end{bmatrix}
+```
+
+\end{bmatrix}
+$$
+
+```elixir
+Nx.argmin(t, axis: 3)
+```
+
+#### Option `:tie_break`
+
+You have the `:tie_break` option to decide how to operate with you have several occurences of the result. It defaults to `tie_break: :low`.
+
+```elixir
+t4 = ~V[2 0 0 0 1]
+
+%{
+  argmin_with_default: Nx.argmin(t4) |> Nx.to_number(),
+  argmin_with_tie_break_high_option: Nx.argmin(t4, tie_break: :high) |> Nx.to_number()
+}
+```
+
+#### Option `:keep_axis`
+
+Its default value is `false` but when this option set to `keep_axis: true`, you reshape the resulting tensor with a new shape where the dimension of the working axis to $1$. For example, `Nx.argmax(t, axis: :y, keep_axix: true)` will have a shape a `[x: 2,y: 1,z: 2, t: 3]`. This mean that:
+
+```elixir
+Nx.argmin(t, axis: 1, keep_axis: true) ==
+  Nx.argmin(t, axis: 1) |> Nx.reshape({2, 1, 2, 3}, names: [:x, :y, :z, :t])
+```
+
+$$
+\begin{bmatrix}
+  \begin{bmatrix}
+
+```
+& 0, 2, 1 & \\
+& 2, 0,1 &
+```
+
+\end{bmatrix}
+  \\ 
+  \begin{bmatrix}
+
+```
+& 1, 2,0 & \\
+& 2,0,0 &
+```
+
+\end{bmatrix}
+\end{bmatrix}
+
+\to
+
+\begin{bmatrix}
+  \begin{bmatrix}
+
+```
+  \begin{bmatrix}
+    & 0, 2, 1 & \\
+    & 2, 0,1 &
+  \end{bmatrix}
+```
+
+\end{bmatrix}
+  \\ 
+  \begin{bmatrix}
+
+```
+  \begin{bmatrix}
+    & 1, 2,0 & \\
+    & 2,0,0 &
+  \end{bmatrix}
+```
+
+\end{bmatrix}
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+#### Multi-row aggregation
+
+<!-- livebook:{"break_markdown":true} -->
+
+Suppose you want to aggregate along the axis `x` and `z`. Then you should get a tensor of shape `[y: 3, t: 3]`. Given an index `x` and `z`, you aggregate all the numbers with the same indexes $(y,t)$.
+
+$$
+r[y][t] = \rm{agg}\big( t0[y]0[t], t0[y]1[t], t1[y]0[t], t1[y]1[t]\big)
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+\begin{bmatrix} x=0
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+   \boxed{1}, 2, 3 & \\
+   \boxed{1}, -2, 3 &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix} 
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 5, -1, 3 &\\
+  & -5, 1, 3 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+  \\
+  x=1
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+   \boxed{4}, 6, 1 & \\
+   \boxed{4},-6, 1 &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix}
+  & 1, 2, 3 & \\
+  & 1, -2, 3 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+\end{bmatrix}
+
+\to
+
+\begin{bmatrix}
+ \rm{agg}(1,1,4,4),\rm{agg}(2,-2,6,-6),\rm{agg}(3,3,1,1) &\\
+  \rm{agg}(4,-4,1,1), \rm{agg}(-3,3,2,-2), \rm{agg}(2,2,3,3) & \\
+  \rm{agg}(5,-5,4,-4) , \rm{agg}(-1,1,-3,3,  \rm{agg}(3,3,2,2)
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+From this, it is easy to undertand what the aggregate returns:
+
+```elixir
+Nx.reduce_min(t, axes: [0, 2])
+```
+
+```elixir
+Nx.reduce_max(t, axes: [0, 2])
```

Comment:
```suggestion
Nx.reduce_max(t, axes: [:x, :t])
```

Reply:
+ Nx.reduce_max(t, axes: [:x, :z])

Reply:
This was intentional so we have a greater variety of examples. Sorry for not explaining

Reply:
ah ok :)